packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1634,7 +1651,9 @@ async function renderToStream(
 
   let reactServerResult: null | ReactServerResult = null
 
-  const setHeader = res.setHeader.bind(res)
```

Comment:
we can just use `res.appendHeader` insteads for React's onHeaders callback. I'd remove the other changes. appending is not always useful. for instance it doesn't make sense for location. If you want to resubmit the PR with just `appendHeader` for the onHeaders callback I think we can merge quickly

Reply:
Many thanks for the review @gnoff!

I've refactored the fix to use `res.appendHeader`  now instead and reverted the other changes (https://github.com/vercel/next.js/pull/73431/commits/51fd66b5661738499f244a76cf4ac22bc68639bc).

I've extracted the part of setting `metadata.headers` to a shared function to be used for both `setHeader` and `appendHeader`—hope that makes sense!

Let me know if there's something else left to do! I've also enabled edits by maintainers if there's something quick you'd like to adjust.

Reply:
Some seemingly unrelated tests failed in the last CI run ([example](https://github.com/vercel/next.js/actions/runs/12443552847/job/34742902118?pr=73431)), any chance the tests are a bit flaky?